AdminWebsite/AdminWebsite/ClientApp/src/app/booking/add-participant/add-participant.component.ts

```diff
@@ -802,4 +830,113 @@ export class AddParticipantComponent extends BookingBaseComponent implements OnI
         const findHearingRole = partyHearingRoles.hearingRoles.find(x => x.name === hearingRole);
         return findHearingRole && findHearingRole.userRole === 'Representative';
     }
+
+    handleContinueRemoveInterpreter() {
+        this.showConfirmRemoveParticipantInterpretedFor = false;
+        this.removeParticipantAndInterpreter();
+        this.populateInterpretedForList();
+    }
+    handleCancelRemoveInterpreter() {
+        this.showConfirmRemoveParticipantInterpretedFor = false;
+    }
+    get interpreterForInvalid() {
+        return this.interpreterFor.invalid && (this.interpreterFor.dirty || this.interpreterFor.touched || this.isShowErrorSummary);
+    }
+    private populateInterpretedForList() {
+        const interpreteeHearingRolesList: Array<string> = [
+            HearingRoles.LITIGANT_IN_PERSON,
+            HearingRoles.WITNESS,
+            HearingRoles.APP,
+            HearingRoles.MACKENZIE_FRIEND
+        ];
+        this.interpreteeList = this.hearing.participants.filter(item =>
+            interpreteeHearingRolesList.includes(item.hearing_role_name.toLowerCase().trim())
+        );
+        const interpreterForModel = Object.assign(new ParticipantModel(), {
+            id: this.constants.PleaseSelect,
+            first_name: this.constants.PleaseSelect,
+            last_name: '',
+            email: this.constants.PleaseSelect
+        });
+        this.interpreteeList.unshift(interpreterForModel);
+    }
+    private setInterpreterForValidation() {
+        if (this.isRoleInterpreter(this.role.value)) {
+            this.interpreterFor.setValidators([Validators.required, Validators.pattern(Constants.PleaseSelectPattern)]);
+            this.interpreterFor.updateValueAndValidity();
+            this.isInterpreter = true;
+        } else {
+            this.interpreterFor.clearValidators();
+            this.interpreterFor.updateValueAndValidity();
+            this.interpreterFor.setValue(Constants.PleaseSelect);
+            this.isInterpreter = false;
+        }
+    }
+    private isRoleInterpreter(hearingRole: string): boolean {
+        return hearingRole.trim().toLowerCase() === HearingRoles.INTERPRETER.toLowerCase();
+    }
+    private hearingHasAnInterpreter(): boolean {
+        const hearingHasInterpreter = this.hearing.participants.find(
+            p => p.hearing_role_name.toLowerCase().trim() === HearingRoles.INTERPRETER.toLowerCase()
+        );
+        return hearingHasInterpreter ? true : false;
```

Comment:
You can simplify this like this: `return !!hearingHasInterpreter`

It may also be better to use `.some()` as this returns a boolean. Then you don't need the above and can just return `hearingHasInterpreter`

Reply:
update to use some()